dashboard/app/controllers/api/v1/pd/workshops_controller.rb

```diff
@@ -15,7 +15,18 @@ def index
       @workshops = @workshops.organized_by(current_user)
     end
 
-    render json: @workshops, each_serializer: Api::V1::Pd::WorkshopSerializer
+    if params[:include_enrollments]
+      enrollments = Pd::Enrollment.where(email: current_user.email)
+      return_val = []
+
+      @workshops.each do |workshop|
+        return_val << Api::V1::Pd::WorkshopSerializer.new(workshop, scope: enrollments.find_by(pd_workshop_id: workshop.id)).attributes
```

Comment:
To leave this open to future additions, might it be better to save the scope as a hash? `scope: {enrollment_id: enrollment.try(&:id)}`